test/adapter/json/nested_relationships_test.rb

```diff
@@ -0,0 +1,248 @@
+require 'test_helper'
+
+NestedPostSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :title
+
+  has_many :comments, include: :author
+end
```

Comment:
@joaomdmoura I'd really like to see us get away from this type of class definition.  If we need it to be this way, I'd say that's a smell.

The only times I think we should build a class like this are:

1. Sentinal objects or Exception subclasses because it's shorter
1. Structs because you have to
1. meta programming
1. testing which probably includes some combination of the above.